rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta

```diff
@@ -46,13 +47,93 @@ type MasterAgreementVariableSet: <"Defines a type where additional variables ass
+
+type Representations:
+
+type IndexAdjustmentEvents: <"Defines the specification of the consequences of Index Events as defined by the 2002 ISDA Equity Derivatives Definitions.">
+
+type FxAdditionalTerms: <"TransactionAdditionalTerms which apply to the CurrencyPair asset class.">
+
+type DeterminationRolesAndTerms: <"Defines the roles and related terms which document the agreement of parties about any determination requirements ; mostly about Extraordinary Events, without being necessarily restricted to such scope, as further specified in the particular product at stake e.g. for instance when Calculation Agent is mentioned as the Price Determination Method enumarated value, etc.">
+
+type AdditionalDisruptionEvents: <"A type for defining the Additional Disruption Events.">
+
+    changeInLaw boolean (0..1) <"Per 2002 ISDA Equity Derivatives Definitions: ">
+    failureToDeliver boolean (0..1) <"Per 2002 ISDA Equity Derivatives Definitions">
+    insolvencyFiling boolean (0..1) <"Per 2002 ISDA Equity Derivatives Definitions">
+    hedgingDisruption boolean (0..1) <"Per 2002 ISDA Equity Derivatives Definitions">
+    increasedCostOfHedging boolean (0..1) <"Per 2002 ISDA Equity Derivatives Definitions">
+    foreignOwnershipEvent boolean (0..1) <"Per ISDA Def ">
+    lossOfStockBorrow boolean (0..1) <"Per 2002 ISDA Equity Derivatives Definitions:">
+    maximumStockLoanRate number (0..1) <"Specifies the maximum stock loan rate for Loss of Stock Borrow. A percentage of 5% is represented as 0.05.">
+    increasedCostOfStockBorrow boolean (0..1) <"Per 2002 ISDA Equity Derivatives Definitions">
+    initialStockLoanRate number (0..1) <"Specifies the initial stock loan per ISDA Def. A percentage of 5% is represented as 0.05.">
+    determiningParty AncillaryRoleEnum (0..1) <"Specifies the party which determines additional disruption events.">
+    additionalBespokeTerms Clause (0..*) <"Where parties may optionnaly describe any extra bespoke agreements, in regards of the standardized Extraordinary Events.">
+    condition MaximumStockLoanRate: <" FpML specifies the maximumStockLoanRate as a RestrictedPercentage, meaning that its value is comprised between 0 and 1.">
+        if maximumStockLoanRate exists then
+            maximumStockLoanRate >= 0 and maximumStockLoanRate <= 1
+
+    condition InitialStockLoanRate: <" FpML specifies the initialStockLoanRate as a RestrictedPercentage, meaning that its value is comprised between 0 and 1.">
+        if initialStockLoanRate exists then
+            initialStockLoanRate >= 0 and initialStockLoanRate <= 1
+
+    condition DisruptionEventsDeterminingParty:
+        if determiningParty exists then
+            determiningParty = AncillaryRoleEnum -> DisruptionEventsDeterminingParty
+
+type Clause: <"A type for documenting additional clause that cannot yet be represented with the model and yet needed for a digital representation of the agreement">
```

Comment:
This type is very similar to the recently added MasterAgreementClause and associated types. I understand that the changes you are contributing here were done a while ago, before the MasterAgreementClause type was added to the model, but it does look as if there is significant commonality here.

Reply:
Yes. Clause is the base type that MasterAgreementClause should extend from. MasterAgreementClause is not generic enough to be reusable here. I have shared recommendation for improvement in an email.

Reply:
Clause was not available in the model when MasterAgreementClause was added. I saw your email with suggestions about the Master Agreement types. If you want MasterAgreementClause to extend Clause then I think we will need to investigate and resolve this as part of this PR.

Reply:
As agreed, we will explore enhancements for masterAgreementClause seperately.